### PR TITLE
Addition of email service functionality to email.q 

### PR DESCRIPTION
**Pull request by CormacRoss**
Update email.q , redmine: Feature #2308

### CODE REVIEW

File: docs/utilities.md [hunk 1]
```diff
@@ -394,6 +394,17 @@ debug should be set to 2i (verbose) to extract the full information.
     q).email.test `$"test@aquaq.co.uk"
     ...
 
+A further function .email.sendviaservice can be used to send an email using the default mail server on a separate specified process and can be used to allow latency sensitive processes to offload this piece of functionality. 
+
+The function takes two parameters a process and a dictionary which should follow  the same format as .email.send.The function uses the .async.postback Utility to send the email on the specified process which will return 1b if the async request has been sent and returns the result to the .email.servicecallback function:
+ 
+```
+q).email.sendviaservice[`emailservice;`to`subject`body!(`$"cormac.ross@aquaq.co.uk";"test email";("hi";"this is an email from torq"))]
```
Comment (ThomasSmyth) [inline docs/utilities.md]: Put a fake address here, something like `user@business.com`

File: docs/utilities.md [hunk 2]
```diff
@@ -394,6 +394,17 @@ debug should be set to 2i (verbose) to extract the full information.
...
 
+A further function .email.sendviaservice can be used to send an email using the default mail server on a separate specified process and can be used to allow latency sensitive processes to offload this piece of functionality. 
+
+The function takes two parameters a process and a dictionary which should follow  the same format as .email.send. The function uses the .async.postback Utility to send the email on the specified process which will return 1b if the async request has been sent and returns the result to the .email.servicecallback function:
+ 
+```
+q).email.sendviaservice[`emailservice;`to`subject`body!(`$"test@aquaq.co.uk";"test email";("hi";"this is an email from torq"))]
+1b
+q)2019.01.04D12:02:57.641940000|homer.aquaq.co.uk|gateway|gateway1|INF|email|Email sent successfully
+
```
Comment (jonnypress) [inline docs/utilities.md]: Could you expand a little more on how the callback mechanism works

File: docs/utilities.md [hunk 3]
```diff
@@ -394,6 +394,17 @@ debug should be set to 2i (verbose) to extract the full information.
     q).email.test `$"test@aquaq.co.uk"
     ...
 
+A further function .email.sendviaservice can be used to send an email using the default mail server on a separate specified process and can be used to allow latency sensitive processes to offload this piece of functionality. 
+
+The function takes two parameters a process and a dictionary which should follow  the same format as .email.send. The function uses the .async.postback Utility to send the email by calling .email.servicesend on the specified process. The postback function immediately returns a success vector indicating that the the async request has been sent and when the function has been run on the server the results are posted back to the client function  email.servicecallback which logs the email status.
```
Comment (jonnypress) [inline docs/utilities.md]: "success vector" should be "success value" or "success boolean" I think

File: docs/utilities.md [hunk 4]
```diff
@@ -394,6 +394,17 @@ debug should be set to 2i (verbose) to extract the full information.
...
+ 
+```
+q).email.sendviaservice[`emailservice;`to`subject`body!(`$"test@aquaq.co.uk";"test email";("hi";"this is an email from torq"))]
+1b
+q)2019.01.04D12:02:57.641940000|homer.aquaq.co.uk|gateway|gateway1|INF|email|Email sent successfully
```
Comment (jonnypress) [inline docs/utilities.md]: remove hostname

**Pull request merged**